v2/pkg/engine/datasource/graphql_datasource/graphql_datasource.go

```
@@ -1274,6 +1284,49 @@ func (p *Planner[T]) printQueryPlan(operation *ast.Document) {
 	p.debugPrintln(args...)
 }
 
+func (p *Planner[T]) generateQueryPlansForFetchConfiguration(operation *ast.Document) {
+	if !p.includeQueryPlanInFetchConfiguration {
+		return
+	}
+	query, err := astprinter.PrintStringIndent(operation, p.upstreamSchema, "  ")
+	if err != nil {
+		return
```

should we also propagate this error?

---

yes